tests/test_notebooks_run.py

```diff
@@ -2,30 +2,34 @@
 collect all notebooks in examples, and check that they run without error
 """
 
+from __future__ import annotations
+
+from pathlib import Path
+
 import nbformat
 from nbconvert.preprocessors import ExecutePreprocessor
-from pathlib import Path
-from glob import glob
-from matplotlib import pyplot as plt  # this script doesn't use this but the notebooks do
+
 this_file_loc = Path(__file__).parent
 
-def check_notebook_runs(notebook_loc):
 
-    print(f'running: {notebook_loc}...')
+def check_notebook_runs(notebook_loc: Path):
+    print(f"running: {notebook_loc}...")
     try:
-        with open(notebook_loc, encoding='utf8') as f:
+        with notebook_loc.open(encoding="utf8") as f:
             nb = nbformat.read(f, as_version=4)
-        ep = ExecutePreprocessor(timeout=600, kernel_name='python3')
-        ep.preprocess(nb, {'metadata': {'path': Path(notebook_loc).parent}})
-    except Exception as e:
-        raise Exception(f'failed to run notebook {notebook_loc}')
-    print(f'success!')
+        ep = ExecutePreprocessor(timeout=600, kernel_name="python3")
+        ep.preprocess(nb, {"metadata": {"path": Path(notebook_loc).parent}})
+    except Exception as exc:
+        error_msg = f"failed to run notebook {notebook_loc}"
+        raise Exception(error_msg) from exc  # noqa: TRY002 # FIXME
```

Comment:
Could be a `RuntimeError`, if you prefer. The point here is to just make sure the tests fail.

Reply:
In my opinion, this `try... except` doesn't mean much.
It seems better to pass `notebook` as a parameter (using `pytest.mark.parametrize`).

Reply:
> In my opinion, this `try... except` doesn't mean much. It seems better to pass `notebook` as a parameter (using `pytest.mark.parametrize`).

@till-m 
Should I fix it in this pr?

Reply:
Yes, please!